File: day_4/puzzle_2/solve.py
import os
import re
import sys

sys.path.append(
    os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
)

from benchmark.benchmark import benchmark


def main():
    with open("./day_4/input.txt", "r") as file:
        pairs = file.read().splitlines()

        count = 0

        for pair in pairs:
            start_a, end_a, start_b, end_b = map(int, re.split(r"[, -]", pair))

            if start_a <= end_b and end_a >= start_b:
                count += 1

        return count


if __name__ == "__main__":
    benchmark(main, "day_4_puzzle_2")

File: day_4/puzzle_2/solve.rb
# frozen_string_literal: true

require_relative '../../benchmark/benchmark'

Benchmark.run('day_4_puzzle_2') do
  File.open('./day_4/input.txt', 'r') do |file|
    pairs = file.read.split("\n")

    count = pairs.count do |pair|
      start_a, end_a, start_b, end_b = pair.split(/[, -]/).map(&:to_i)

      if start_a <= end_b && end_a >= start_b
        true
      else
        false
      end
    end

    count
  end
end
